util/runtests.sh

```diff
@@ -132,6 +132,56 @@ fi
 # try to clean up a bit
 sudo odemis-stop
 
+echo -e "\n\n==============================================="
+echo "Running unit tests in python 3"
+echo "Running unit tests on $(date)" > "$TESTLOG"
+
+# run each test script and save the output
+failures=0
+for f in $testfiles; do
+    echo "Running $f..."
+    if ! grep -q "__main__" $f; then
+        echo "WARNING: test $f seems to not be runnable"
+    fi
+    echo "Running $f:" >> "$TESTLOG"
+    prev_size=$(wc -l < "$TESTLOG")
+    # run it in its own directory (sometimes they need specific files from there)
+    pushd "$(dirname $f)" > /dev/null
+        # Automatically kill after MAXTIME, then try harder after 30 s
+        timeout -k 30 $MAXTIME python3 $f --verbose >> "$TESTLOG" 2>&1
+        status=$?
+        echo $f returned $status >> "$TESTLOG" 2>&1
+    popd > /dev/null
+    # Don't show test output if the file hasn't grown, as it'd be the previous test output
+    new_size=$(wc -l < "$TESTLOG")
+    if [[ "$new_size" == "$prev_size" ]]; then
+        echo "NOT RUN"
+    else
+        tail -n "+$prev_size" "$TESTLOG" | grep -E 'OK' | tail -1
+        tail -n "+$prev_size" "$TESTLOG" | awk "/^FAIL: /,/FAILED/"
+        tail -n "+$prev_size" "$TESTLOG" | awk "/^ERROR: /,/FAILED/"
+        #tail -n "+$prev_size" "$TESTLOG" | awk '/===/, /FAILED/'
+    fi
+    echo -e "\n"
+    if [ "$status" -gt 0 ]; then
+        # TODO: failures can increase even if the test reported OK, if it was killed
+        # => synchronise it with FAILED
+        failures=$(( $failures + 1 ))
+    fi
+
+    # Stops the back-end, just in case it happens to still be running
+    sudo odemis-stop
+done
+
+if [ $failures -gt 0 ]; then
+    echo "$failures test failed. See $TESTLOG for error messages."
+else
+    echo "All tests passed"
+fi
+
+# try to clean up a bit
+sudo odemis-stop
```

Comment:
It's a lot of duplicate code. I think it should be fine to move that code to a function (and use all the variables as globals). You could then just have one single variable PYTHON_INTERPRETER to update, or pass it as an argument.